src/lib/journey/callbacks/kba/kba-create.svelte

```diff
@@ -18,18 +18,47 @@
    * Details: Each callback is wrapped by the SDK to provide helper methods
    * for accessing values from the callbacks received from AM
    ************************************************************************* */
-  const prompt = callback.getPrompt();
-  const questions = callback
-    .getPredefinedQuestions()
-    ?.map((label, idx) => ({ text: label, value: `${idx}` }));
+  const inputArr = callback?.payload?.input;
   const inputName = callback?.payload?.input?.[0].name || `kba-${idx}`;
   const inputNameQuestion = inputName;
-  const inputArr = callback?.payload?.input;
   const inputNameAnswer = Array.isArray(inputArr) && inputArr[1].name;
+  const prompt = callback.getPrompt();
+  const questions = callback
+    .getPredefinedQuestions()
+    ?.map(
+      (label, idx) => ({ text: label, value: `${idx}` }),
+    );
 
+  let customQuestionIndex: string | null = null;
+  let displayCustomQuestionInput = false;
+  let shouldAllowCustomQuestion: boolean | undefined;
   let value = '';
 
-  callback.setQuestion(questions[0].text);
+  /**
+   * `getOutputValue` throws if it doesn't find this property. There _may_ be a context
+   * in which the property doesn't exist, so I'm going to wrap it in a try-catch, just
+   * in case
+  */
+  try {
+    shouldAllowCustomQuestion = callback.getOutputValue('allowUserDefinedQuestions') as boolean;
```

Comment:
The KBA callback has an option to allow custom questions. So, most of this new code is to allow this feature. 

Being that I'm using this low level callback feature (we really should add a higher level API for this in the SDK's KbaCallback handler), it throws when this `allowUserDefinedQuestions` doesn't exist, so in order to keep the widget from exploding I'm wrapping it in a try-catch.

Reply:
Oh, I'm not sure if it's possible for the KBA callback to NOT have the `allowUserDefinedQuestions` property, but I have quite a few example KBA callbacks around that did indeed NOT have it, so that's why I'm adding the try-catch, just in case.